### PR TITLE
Apply fix to issue pulling the coredns image for 1.24

### PR DESCRIPTION
**Pull request by brianmcarey**
The patch bump for 1.24 is failing due to an issue pulling the coredns
image[1]. 
This change prepulls the image and tags it as kubeadm expects. A
similar update was made to the 1.25 provider[2].

it also includes the patch bump PR changes so merging this can close that PR

Closes https://github.com/kubevirt/kubevirtci/pull/928

/cc @dhiller @enp0s3 

[1] https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirtci/928/check-provision-k8s-1.24/1618672693480001536
[2] https://github.com/kubevirt/kubevirtci/commit/25ce23aa0ba9da19ece1a8e2189da6c283e8b27d

### CODE REVIEW

**Review by dhiller — APPROVED** [review]
/approve

**Comment by brianmcarey** [comment]
/hold 

I think an update is needed to add the image tag to the extra prepull images list.
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirtci/953/check-provision-k8s-1.24/1619079238701289472#1:build-log.txt%3A6201

**Comment by brianmcarey** [comment]
/unhold 

@dhiller can you please take a look again?

**Review by dhiller — APPROVED** [review]
/approve

File: cluster-provision/k8s/1.24/k8s_provision.sh [hunk 1]
```diff
@@ -160,6 +160,10 @@ envsubst < $kubeadm_raw > $kubeadm_manifest
 
 until ip address show dev eth0 | grep global | grep inet6; do sleep 1; done
 
+
+# prepull coredns:v1.8.6 and retag to expected tag
+podman pull registry.k8s.io/coredns/coredns:v1.8.6 && podman tag registry.k8s.io/coredns/coredns:v1.8.6 k8s.gcr.io/coredns:v1.8.6
```
Comment (oshoval) [inline cluster-provision/k8s/1.24/k8s_provision.sh:165]: can we derive the tag from the manifests ? so once it is changed we dont need to update this code?
(unless it is just a WA and expected to be dropped soon)

**Comment by dhiller** [comment]
/override check-provision-k8s-1.26 

we want a new provider set for 1.24 because previous was published with broken cri-o version

**Review by dhiller — APPROVED** [review]

**Pull request merged**